qiskit/validation/jsonschema/schema_validation.py

```diff
@@ -131,21 +131,38 @@ def validate_json_against_schema(json_dict, schema,
         SchemaValidationError: Raised if validation fails.
     """
 
-    try:
-        if isinstance(schema, str):
-            schema_name = schema
-            schema = _SCHEMAS[schema_name]
-            validator = _get_validator(schema_name)
-            validator.validate(json_dict)
-        else:
+    if isinstance(schema, str):
+        schema_name = schema
+        schema = _SCHEMAS[schema_name]
+        validator = _get_validator(schema_name)
+        errors = list(validator.iter_errors(json_dict))
+        if errors:
+            best_match_error = jsonschema.exceptions.best_match(errors)
+            failure_path = list(best_match_error.absolute_path)
+            if len(failure_path) > 1:
+                failure_path = failure_path[:-1]
+            error_path = ""
+            for component in failure_path:
+                if isinstance(component, int):
+                    error_path += "[%s]" % component
+                else:
+                    error_path += "['%s']" % component
+            if failure_path:
+                err_message = "Validation failed. Possibly at %s" % error_path
+                err_message += " because of %s" % best_match_error.message
+            else:
+                err_message = "Validation failed. "
+                err_message += "Possibly because %s" % best_match_error.message
+            raise SchemaValidationError(err_message)
+    else:
+        try:
             jsonschema.validate(json_dict, schema)
-    except jsonschema.ValidationError as err:
-        if err_msg is None:
-            err_msg = "JSON failed validation. Set Qiskit log level to DEBUG " \
-                      "for further information."
-        newerr = SchemaValidationError(err_msg)
-        newerr.__cause__ = _SummaryValidationError(err)
-        logger.debug('%s', _format_causes(err))
+        except jsonschema.ValidationError as err:
+            if err_msg is None:
+                err_msg = ("JSON failed validation. Set Qiskit log level to "
+                           "DEBUG for further information.")
+            newerr = SchemaValidationError(err_msg)
+            newerr.__cause__ = _SummaryValidationError(err)
         raise newerr
```

Comment:
Not blocking but would this be more python if it was
```python
raise newerr from _SummaryValidationError(err)
```
?

Reply:
Sure, I was just indenting what was there before to minimize the changes. I can update it if needed though.